docker-compose.yml

```diff
@@ -1,40 +1,49 @@
 version: "3.8"
 services:
-  api:
-    container_name: api
-    restart: on-failure
+  oracc-api:
+    container_name: oracc-api
+    restart: always
     build:
       context: .
       dockerfile: ./compose/api/Dockerfile
     ports:
-      - "8000:8000"
+    - ${ORACC_PORT:-8000}:8000
     stdin_open: true
     tty: true
     environment:
-      - ELASTICSEARCH_HOST=http://elasticsearch:9200
+    - ELASTICSEARCH_HOST=http://elasticsearch:9200
     depends_on:
-      - elasticsearch
+    - elasticsearch
 
   elasticsearch:
+    restart: always
     build:
       context: ./compose/elasticsearch
       dockerfile: Dockerfile
     environment:
-      - discovery.type=single-node # this is for development only
-      - xpack.security.enabled=false # this should be set to true in production
-      - http.cors.enabled=true
-      - http.cors.allow-origin=/http:\/\/(elasticsearch|localhost):9200/
-    healthcheck:
-      test: ["CMD", "curl", "-f", "http://localhost:9200/_cluster/health?wait_for_status=yellow&timeout=3s"]
-      interval: 4s
-      timeout: 4s
-      retries: 5
-      start_period: 5s
+    - discovery.type=single-node
+    - xpack.security.enabled=false
+    - http.cors.enabled=true
+    - http.cors.allow-origin=/http:\/\/(elasticsearch|localhost):9200/
+    volumes:
+    - es_data:/usr/share/elasticsearch/data:rw
 
-  ingest-service:
+  oracc-ingest:
+    container_name: oracc-ingest
+    restart: "no"
     build:
       context: .
       dockerfile: ./compose/ingest/Dockerfile
+    volumes:
+    - ingest:/glossaries:ro
     depends_on:
-      elasticsearch:
-        condition: service_healthy
+    - elasticsearch
```

Comment:
I'm guessing you removed the healthchecks in favour of your alternative custom approach in `ingest/bulk_upload.py`? Is there a reason why this approach is not recommended?

Reply:
Advantages of health check:
* you can take advantage of docker's own retrying mechanism, so it's less code
Advantages of doing it in Python:
* we can use the Python wrapper (thanks for getting me to seek this out, although it turns out we have to do the retrying ourselves)
* we can use it outside of docker if we want (although YAGNI)
* we don't have to worry about what happens at restart, which I think I might have got wrong in the old version but it's really hard to test, whereas this one pretty much tests itself.